src/components/chip/chip.ts

```diff
@@ -0,0 +1,129 @@
+}
+/**
+ * Chips help people enter information, make selections, filter content, or trigger actions.
+ *
+ * @element igc-chip
+ *
+ * @slot - Renders the chip data.
+ * @slot prefix - Renders content before the data of the chip.
+ * @slot suffix - Renders content after the data of the chip.
+ *
+ * @fires igcRemove - Emits an event when the chip component is removed. Returns the removed chip component.
+ * @fires igcSelect - Emits event when the chip component is selected/deselected and any related animations and transitions also end.
+ *
+ * @csspart base - The base wrapper of the chip.
+ * @csspart prefix - The prefix container of the chip.
+ * @csspart suffix - The suffix container of the chip.
+ */
+@themes({ material, bootstrap, fluent, indigo })
+export default class IgcChipComponent extends EventEmitterMixin<
+  IgcChipEventMap,
+  Constructor<LitElement>
+>(LitElement) {
+  public static readonly tagName = 'igc-chip';
+
+  public static styles = styles;
+
+  /** Sets the disabled state for the chip. */
+  @property({ type: Boolean, reflect: true })
+  public disabled = false;
+
+  /** 	Defines if the chip is removable or not. */
+  @property({ type: Boolean, reflect: true })
+  public removable = false;
+
+  /**	Defines if the chip is selectable or not. */
+  @property({ type: Boolean, reflect: true })
+  public selectable = false;
+
+  /**	Defines if the chip is selected or not. */
+  @property({ type: Boolean, reflect: true })
+  @blazorTwoWayBind('igcSelected', 'detail')
+  public selected = false;
+
+  /** A property that sets the size of the chip component. */
+  @property({ reflect: true })
+  public size: 'small' | 'medium' | 'large' = 'medium';
+
+  /** A property that sets the color variant of the chip component. */
+  @property({ reflect: true })
+  public variant!:
+    | 'none'
+    | 'primary'
+    | 'success'
+    | 'danger'
+    | 'warning'
+    | 'info';
+
+  protected handleSelect() {
+    if (this.selectable) {
+      this.selected = !this.selected;
+      this.emitEvent('igcSelect', { detail: this.selected });
+    }
+  }
+
+  protected handleRemove(e: Event) {
+    document.getElementsByName('remove');
+    this.emitEvent('igcRemove');
+    e.stopPropagation();
+  }
+
+  protected override render() {
+    return html`
+      <button
+        part="base"
+        .disabled="${this.disabled}"
+        aria-selected="${this.selected ? 'true' : 'false'}"
+        aria-disabled="${this.disabled ? 'true' : 'false'}"
+        @click=${this.handleSelect}
+      >
+        <span part="prefix">
+          ${this.selectable && this.selected
+            ? html`<slot @slotchange=${this.slotChanges} name="select">
+                <igc-icon size=${this.size} name="select"></igc-icon>
+              </slot>`
+            : ''}
+          <slot name="start"></slot>
+        </span>
+        <slot></slot>
+        <span part="suffix">
+          <slot name="end"></slot>
+          ${this.removable && !this.disabled
```

Comment:
Why do we have to hide the remove icon when the component is being disabled? This will result in UI jumping when programmatically changing the disabled property.

Reply:
@StefanIvanov please consider my comment.

Reply:
Removing a disabled chip will never be something that the user will be able to execute, so I prefer not to take up space for redundant things and save that space to e.g. show more chips on a line. I don't expect toggling disabled state to be that often so that jumping becomes that obvious.